User's navigation

leonplondon opened the pull request:
# Summary
Implemented basic navigation on application.

# Trello Card
https://trello.com/c/ButF5txO/18-u1-frontend-login-navegaci%C3%B3n

leonplondon committed:
Renamed application class from bootstrap indications.
Login UI markup.

leonplondon committed:
Renamed application class from bootstrap indications.
Login UI markup.

leonplondon committed:
Added validations to login form.
User's credentials are persisted on SharedPreferences.

leonplondon committed:
Added validations to login form.
User's credentials are persisted on SharedPreferences.

leonplondon committed:
Added validations to login form.
User's credentials are persisted on SharedPreferences.

leonplondon committed:
Added validations to login form.
User's credentials are persisted on SharedPreferences.

leonplondon committed:
Added validations to login form.
User's credentials are persisted on SharedPreferences.

leonplondon committed:
Removed unused code

leonplondon committed:
Applied review requested changes

leonplondon committed:
Applied review requested changes

leonplondon committed:
Applied review requested changes

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Added user's navigation flow.
User can access home screen.
User can go to terms page.
User can access sign up screen.

leonplondon committed:
Merge branch 'login-frontend-3' of github.com:wolox-training/ldp-android into login-frontend-3

leonplondon committed:
Added nullabities annotations

leonplondon committed:
Fixed typo on annotation

leonplondon committed:
Fixed typo on annotation

leonplondon committed:
Fixed typo on annotation

leonplondon committed:
Fixed merge conflicts from rebase

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/customViews/TermsAndConditions.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ *
+ */
+public class TermsAndConditions extends LinearLayout {
```
There is no need to do this. You can set your string.xml with the text like 
 **<string name="fragment_login_terms_conditions">By joining you agree to our <a href="http://www.wolox.com.ar/">Terms &amp; Conditions</a></string>**

And then in your fragment use **mTermsAndConditions.setMovementMethod(LinkMovementMethod.getInstance());**

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/home/MainActivity.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * Application home;
+ */
+public class MainActivity extends WolmoActivity {
```
Maybe we can change the name to something like HomeActivity. What do you think?

leonplondon replied:
I think is better HomeActivity because the main (entry) activity for the App is LoginActivity

leonplondon committed:
Login validations (#2)

* Renamed application class from bootstrap indications.
Login UI markup.

* Added validations to login form.
User's credentials are persisted on SharedPreferences.

* Login UI markup (#1)

* Renamed application class from bootstrap indications.
Login UI markup.

* Fixed gradient view properties(width).
Removed unused code on AndroidManifest.xml

* Renamed application class from bootstrap indications.
Login UI markup.

* Added validations to login form.
User's credentials are persisted on SharedPreferences.

* Applied review requested changes

* Added nullabities annotations

* Fixed typo on annotation

* Fixed PR # 2

leonplondon committed:
Fixed merge conflicts

leonplondon committed:
Fixed PR#3

leonplondon committed:
Fixed PR#3

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/signup/SignUpFragment.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public void setUi(View v) {
+        super.setUi(v);
+        Log.d("ldpenal", "setUi: ");
```
Should remove this or change the TAG 😄

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/utils/StringUtils.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public final class StringUtils {
+
+    static final String EMAIL_REGEX = "[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)" +
```
Rebase this repo until merge, to remove this file

DemianIac commented on app/src/main/res/layout/fragment_login.xml:
```diff
@@ -17,7 +17,7 @@
 
     <View
-        android:layout_width="match_parent"
+        android:layout_width="0dp"
```
This should be fixed with the rebase.

DemianIac reviewed (approved):
Just remove one log and remember to do the rebase. Nice work!!

leonplondon committed:
Fixed PR#3

leonplondon committed:
Closed PR#3

leonplondon committed:
Fixed merge conflict
